### PR TITLE
Fix and improve lobby custom rules warning.

### LINKED ISSUE #11068: Official maps displaying "modded map, experience will change etc."
Was on a server AoAGeneral was hosting and all the official maps came up with the modded map warning.

### PR DESCRIPTION
**Pull request by pchote**
Fixes #11068, and improves the check so that maps with custom lighting and/or weather effects don't trigger the warning.

### CODE REVIEW

File: OpenRA.Game/GameRules/Ruleset.cs [hunk 1]
```diff
@@ -200,5 +200,55 @@ public static Ruleset LoadDefaultsForTileSet(ModData modData, string tileSet)
 
 			return ruleset;
 		}
+
+		static bool AnyCustomYaml(MiniYaml yaml)
```
Comment (Phrohdoh) [inline OpenRA.Game/GameRules/Ruleset.cs]: I don't think checking for any children relates to 'custom' yaml.

Comment (pchote) [inline OpenRA.Game/GameRules/Ruleset.cs]: The nodes that are being checked contain yaml overrides that customize the default rules/weapons/etc.

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

**Comment by obrakmann** [comment]
:+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/35940b2b8eaa2e0d0654c0fa1771fe9f04e550aa)

**Pull request merged**